jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannelState.java

```diff
@@ -511,6 +512,11 @@ public void startAsync(AsyncContextEvent event)
             if (_state != State.HANDLING || (_requestState != RequestState.BLOCKING && _requestState != RequestState.ERRORING))
                 throw new IllegalStateException(this.getStatusStringLocked());
 
+            if (!_failureListener)
+            {
+                _failureListener = true;
+                _servletChannel.getRequest().addFailureListener(this::asyncError);
+            }
```

Comment:
Can't you just add this in the constructor, like the idle timeout listener?

Also, should not there be a similar code in ee9?

Reply:
Only needed if startAsync is called
Added to ee9, but needs your PR to really test it.